server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -1084,7 +1084,7 @@ public void apply(T value, Settings current, Settings previous) {
     }
 
     public static Setting<Version> versionSetting(final String key, final Version defaultValue, Property... properties) {
-        return new Setting<>(key, s -> Integer.toString(defaultValue.externalId), s -> Version.fromId(Integer.parseInt(s)), properties);
+        return new Setting<>(key, s -> Integer.toString(defaultValue.internalId), s -> Version.fromId(Integer.parseInt(s)), properties);
```

Comment:
Version lookup is based on the internal id not the external id See https://github.com/crate/crate/blob/master/server/src/main/java/org/elasticsearch/Version.java#L141